PluginManager/PluginManager.cpp

```diff
@@ -277,26 +276,43 @@ namespace plugin
         return ResolvePathInternal(filename, "PLUGIN_PATH", "plugins/");
     }
 
-    std::filesystem::path PluginManager::ResolvePathInternal(std::string filename,
+    std::filesystem::path PluginManager::ResolvePathInternal(std::filesystem::path filename,
                                                const char* envVarOverride,
                                                std::string defaultPrefix)
     {
         if(filename.empty()) {
             throw std::logic_error{ "Received blank filename in ResolvePathInternal" };
         }
 
-        std::ifstream file(filename.c_str());
```

Comment:
Opening a file to check if it exists isn't great... use `filesystem::exists` instead